mlir/lib/Interfaces/DataLayoutInterfaces.cpp

```diff
@@ -293,6 +293,28 @@ mlir::detail::getDefaultStackAlignment(DataLayoutEntryInterface entry) {
   return value.getValue().getZExtValue();
 }
 
+// Returns the max vector op width if specified in the given entry. If the entry
+// is empty (meaning the spec is missing), returns std::nullopt.
+std::optional<int64_t>
+mlir::detail::getMaxVectorOpWidth(DataLayoutEntryInterface entry) {
+  if (entry == DataLayoutEntryInterface())
```

Comment:
```suggestion
  if (!entry)
```

But is this check useful?